lib/web_ui/lib/src/engine/text_editing.dart

```diff
@@ -305,20 +310,29 @@ class TextEditingElement {
     }
 
     // Subscribe to text and selection changes.
-    _subscriptions
-      ..add(html.document.onSelectionChange.listen(_handleChange))
-      ..add(domElement.onInput.listen(_handleChange));
-
-    // In Firefox, when cursor moves, nor selectionChange neither onInput
-    // events are triggered. We are listening to keyup event to decide
-    // if the user shifted the cursor.
-    // See [SelectionChangeDetection].
+    _subscriptions.add(domElement.onInput.listen(_handleChange));
+
+    /// Detects changes in text selection.
+    ///
+    /// Currently only used in Firefox.
+    ///
+    /// In Firefox, when cursor moves, neither selectionChange nor onInput
+    /// events are triggered. We are listening to keyup event. Selection start,
+    /// end values are used to decide if the text cursor moved.
+    ///
+    /// Specific keycodes are not checked since users/applications can bind
+    /// their own keys to move the text cursor.
+    /// Decides if the selection has changed (cursor moved) compared to the
+    /// previous values.
+    ///
+    /// After each keyup, the start/end values of the selection is compared to the
+    /// previously saved editing state.
     if (browserEngine == BrowserEngine.firefox) {
       _subscriptions.add(domElement.onKeyUp.listen((event) {
```

Comment:
You also mentioned `domElement.onSelect` to fix the select all problem. If that’s coming in a separate PR, then nvm.

Reply:
yes it is coming on a separate PR. I want to make sure all browsers are covered for "Select All" context menu item.